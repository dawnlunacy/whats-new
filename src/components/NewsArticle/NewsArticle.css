.newsArticle {
 background-color: #000222;
 max-width: 500px;
}

.newsImage {
    width: 100%;
}

h3 {
    padding: 10px;
}

footer {
    height: 50px;
    margin: 10px 0px 0px 0px;
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
}
