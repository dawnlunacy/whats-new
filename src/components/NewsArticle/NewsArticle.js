import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
    return (
        <article className="newsArticle"> 
            <img src={props.img} alt="" className="newsImage" width="300" height="200"/> 
            <h3> {props.headline}</h3>
            <p>{props.description}</p>
            <footer>
                <a href={props.url}>Link to article</a> 
            </footer>
        </article>
    )
}

export default NewsArticle;