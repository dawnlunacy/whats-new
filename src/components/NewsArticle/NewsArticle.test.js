import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
    it('should match the NewsArticle snapshot with the data passed to it', () => {
        const article = {
            id: 22,
            headline: " Robbie's Favorite",
            img: "https://i.pinimg.com/originals/df/ad/d7/dfadd7765de8e922206401f0529cae9a.jpg",
            desciption: "Pugs in Sweater Take Over The World",
            url: "https://www.pinterest.nz/pin/575546027375685533/"
        }
        const wrapper = shallow(<NewsArticle 
            img= {article.img}
            headline={article.headline}
            description={article.description}
            url={article.url}/>)

    expect(wrapper).toMatchSnapshot()
    })
})