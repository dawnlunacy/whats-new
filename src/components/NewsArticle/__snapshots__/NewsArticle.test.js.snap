// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the NewsArticle snapshot with the data passed to it 1`] = `
<article
  className="newsArticle"
>
  <img
    alt=""
    className="newsImage"
    height="200"
    src="https://i.pinimg.com/originals/df/ad/d7/dfadd7765de8e922206401f0529cae9a.jpg"
    width="300"
  />
  <h3>
     
     Robbie's Favorite
  </h3>
  <p />
  <footer>
    <a
      href="https://www.pinterest.nz/pin/575546027375685533/"
    >
      Link to article
    </a>
  </footer>
</article>
`;
